### PR TITLE
Add new parameter: repoVisibility for the action: publish:gitea

### PR DESCRIPTION
**Pull request by cmoulliard**
## Hey, I just made a Pull Request!

Add a new parameter `repoVisibility` for the action `publish:gitea` to allow create public or private git repositories on gitea

Template used for the tests with a local gitea server: https://github.com/ch007m/my-backstage-templates/blob/main/gitea/template.yaml

### Results

Screenshot 2024-04-19 at 12 29 00
Screenshot 2024-04-19 at 12 29 05
Screenshot 2024-04-19 at 12 30 11
Screenshot 2024-04-19 at 12 30 20

Issue: https://github.com/backstage/backstage/issues/24377

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by cmoulliard** [comment]
PR is ready to be reviewed @vinzscam @benjdlambert

File: plugins/scaffolder-backend-module-gitea/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@backstage/plugin-scaffolder-backend-module-gitea",
-  "version": "0.1.7",
+  "version": "0.1.8",
```
Comment (benjdlambert) [inline plugins/scaffolder-backend-module-gitea/package.json:3]: Don't need to to this change, the changeset action will do this on release.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Nice! Thanks for this, just a few small nits.

File: plugins/scaffolder-backend-module-gitea/README.md [hunk 2]
```diff
@@ -27,29 +34,29 @@ integrations:
...
+**Important**: As backstage will issue `HTTPS/TLS` requests to the gitea instance, it is needed to configure `gitea` with a valid certificate or at least with a
+self-signed certificate `gitea cert --host localhost -ca` trusted by a CA authority. Don't forget to set the env var `NODE_EXTRA_CA_CERTS` to point to the CA file before launching backstage or to ignore using the following Node.js parameter: `NODE_TLS_REJECT_UNAUTHORIZED=0` !
 
 When done, you can create a template which:
 
-- Declare the `RepoUrlPicker` within the `spec/parameters` section to select the gitea hosts
-- Include a step able to publish by example the newly generated project using the action `action: publish:gitea`
+- Declare the `RepoUrlPicker` within the `spec/parameters` section to select the gitea host and to provide the name of the repository
+- Add an enum list allowing the user to define the visibility about the repository to be created: `public` or `private`. If this field is omitted, `public` is then used by the action.
```
[the remark this answers is not in the document]
Comment (cmoulliard) [reply plugins/scaffolder-backend-module-gitea/README.md:43]: Do you mean that part of this README, the syntax to be used should be: `'public'`, `'private'` AND not `public`, `public` ?

Comment (vinzscam) [inline plugins/scaffolder-backend-module-gitea/README.md:43]: like this 😄

```suggestion
- Add an `enum` list allowing the user to define the visibility about the repository to be created: `public` or `private`. If this field is omitted, `public` is then used by the action.
```

**Comment by cmoulliard** [comment]
Why do we have yarn error as reported by some jobs ? 
```
➤ YN0028: -"@rjsf/utils@npm:5.17.1":
➤ YN0028: -  version: 5.17.1
➤ YN0028: -  resolution: "@rjsf/utils@npm:5.17.1"
➤ YN0028: -  dependencies:
➤ YN0028: -    json-schema-merge-allof: ^0.8.1
➤ YN0028: -    jsonpointer: ^5.0.1
➤ YN0028: -    lodash: ^4.17.21
➤ YN0028: -    lodash-es: ^4.17.21
➤ YN0028: -    react-is: ^18.2.0
➤ YN0028: -  peerDependencies:
➤ YN0028: -    react: ^16.14.0 || >=17
➤ YN0028: -  checksum: 83010de66b06f1046b023a0b7d0bf30b5f47b152893c3b12f1f42faa89e7c7d18b2f04fe2e9035e5f63454317f09e6d5753fc014d43b933c8023b71fc50c3acf
➤ YN0028: -  languageName: node
➤ YN0028: -  linkType: hard
➤ YN0028: -
➤ YN0028: The lockfile would have been modified by this install, which is explicitly forbidden.
➤ YN0000: └ Completed in 0s 216ms
➤ YN0000: Failed with errors in 1s 330ms
```
@benjdlambert

File: plugins/scaffolder-backend-module-gitea/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmoulliard) [reply plugins/scaffolder-backend-module-gitea/README.md]: I highlighted the enum word: 9c65898
I dont have vale errors locally
```
vale --config .vale.ini /Users/cmoullia/code/janus-idp/fork-backstage/plugins/scaffolder-backend-module-gitea/README.md
✔ 0 errors, 0 warnings and 0 suggestions in 1 file.
```

**Comment by benjdlambert** [comment]
You can do a rebase, I fixed this on mainline now.

**Comment by cmoulliard** [comment]
> You can do a rebase, I fixed this on mainline now.

Fixed after rebase @benjdlambert

File: plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts [hunk 3]
```diff
@@ -112,12 +113,23 @@ const createGiteaProject = async (
     This is the default scenario that we support currently
   */
   let response: Response;
+  let visibility: boolean;
```
Comment (benjdlambert) [inline plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts:116]:
```suggestion
  let private: boolean;
```

Comment (benjdlambert) [reply plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts:116]: Think this is a bit clearer for the API call right?

Comment (cmoulliard) [reply plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts:116]: According to typescript "private" private is a reserved word in strict mode. So, can we use it or not ?

`TS1214: Identifier expected. private is a reserved word in strict mode. Modules are automatically in strict mode.`

Comment (benjdlambert) [reply plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts:116]: Maybe `isPrivate` works better then?

Comment (cmoulliard) [reply plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts:116]: I vote for `isPrivate` :-)

Comment (benjdlambert) [reply plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts:116]: Let's go with that then! 😄

File: plugins/scaffolder-backend-module-gitea/README.md [hunk 4]
```diff
@@ -27,29 +34,29 @@ integrations:
       password: '<GITEA_LOCALHOST_PASSWORD>'
 ```
 
-**Important**: As backstage will issue HTTPS/TLS requests to the gitea instance, it is needed to configure `gitea` with a valid certificate or at least with a
-self-signed certificate `gitea cert --host localhost -ca` trusted by a CA authority. Don't forget to set the env var `NODE_EXTRA_CA_CERTS` to point to the CA file before launching backstage !
+**Important**: As backstage will issue `HTTPS/TLS` requests to the gitea instance, it is needed to configure `gitea` with a valid certificate or at least with a
+self-signed certificate `gitea cert --host localhost -ca` trusted by a CA authority. Don't forget to set the env var `NODE_EXTRA_CA_CERTS` to point to the CA file before launching backstage or to ignore using the following Node.js parameter: `NODE_TLS_REJECT_UNAUTHORIZED=0` !
```
Comment (benjdlambert) [inline plugins/scaffolder-backend-module-gitea/README.md:38]:
```suggestion
self-signed certificate `gitea cert --host localhost -ca` trusted by a CA authority. Don't forget to set the env var `NODE_EXTRA_CA_CERTS` to point to the CA file before launching backstage or you can set temporarily `NODE_TLS_REJECT_UNAUTHORIZED=0` but this is not recommended for production!
```

File: plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts (hunk 3 above)
Comment (cmoulliard) [reply plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts:116]: Do you plan to commit the change using `isPrivate` as suggestion or should I do it ?

Comment (benjdlambert) [reply plugins/scaffolder-backend-module-gitea/src/actions/gitea.ts:116]: @cmoulliard I can't push to this branch, so you'd have to do it :)

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Thanks @cmoulliard!

**Pull request merged**